databuilder/query_engines/base_sql.py

```diff
@@ -493,153 +470,163 @@ def round_to_first_of_month(self, date):
     def round_to_first_of_year(self, date):
         raise NotImplementedError
 
-    def apply_aggregates(self, query, aggregate_nodes):
+    def query_to_create_temp_table_from_select_query(
+        self, table: TemporaryTable, select_query: Executable
+    ) -> Executable:
         """
-        For each aggregate node, get the query that will select it with its generated
-        column label, plus the patient id column, and then group by the patient id.
+        Return a query to create `table` and populate it with the results of
+        `select_query`
         """
-        columns = [
-            self.get_aggregate_column(query, aggregate_node)
-            for aggregate_node in aggregate_nodes
-        ]
-        query = query.with_only_columns([query.selected_columns.patient_id] + columns)
-        query = query.group_by(query.selected_columns.patient_id)
+        raise NotImplementedError()
 
-        return query
+    def temp_table_needs_dropping(self, create_table_query: Executable) -> bool:
+        """
+        Given the query used to create a temporary table, return whether the table needs
+        to be explicitly dropped or whether the database will discard it automatically
+        when the session terminates
+        """
+        raise NotImplementedError()
 
-    def get_aggregate_column(self, query, aggregate_node):
+    #
+    # DATABASE CONNECTION
+    #
+    @property
```

Comment:
We can use [`functools.cached_property`](https://docs.python.org/3/library/functools.html#functools.cached_property)